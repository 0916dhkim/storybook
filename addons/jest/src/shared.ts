import type { Parameters } from '@storybook/addons';

// addons, panels and events get unique names using a prefix
export const PARAM_KEY = 'test';
export const ADDON_ID = 'storybookjs/test';
export const PANEL_ID = `${ADDON_ID}/panel`;

export const ADD_TESTS = `${ADDON_ID}/add_tests`;

interface AddonParameters extends Parameters {
  jest?: string | string[] | { disable: true } | null;
}

export function defineJestParameters(parameters: AddonParameters): string[] | null {
  const { jest, fileName: filePath } = parameters;

  console.log(jest);

  if (typeof jest === 'string') {
    return [jest];
  }

  if (jest && Array.isArray(jest)) {
    return jest;
  }

  if (jest === undefined) {
    const fileName = filePath.split('/').pop().split('.')[0];
    return [fileName];
  }

  return null;
}
