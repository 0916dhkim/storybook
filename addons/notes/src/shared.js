// addons, panels and events get unique names using a prefix
export const ADDON_ID = 'storybooks/notes';
export const PANEL_ID = `${ADDON_ID}/panel`;
export const EVENT_ID = `${ADDON_ID}/add_notes`;
