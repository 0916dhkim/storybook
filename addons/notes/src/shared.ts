export const ADDON_ID = 'storybooks/notes';
export const PANEL_ID = `${ADDON_ID}/panel`;
export const PARAM_KEY = `notes`;

export interface API {
  on(event: string, callback: (...args: any) => any): void;
  off(event: string, callback: (...args: any) => any): void;
  emit(event: string, callback: (...args: any) => any): void;

  getParameters(id: string, scope?: string): undefined | Parameters;
}

export type Parameters =
  | string
  | {
      text?: string;
      markdown?: string;
      disable?: boolean;
    };
