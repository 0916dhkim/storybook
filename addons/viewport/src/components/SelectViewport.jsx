import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { viewports, defaultViewport } from './viewportInfo';
import * as styles from './styles';

export class SelectViewport extends Component {
  static propTypes = {
    onChange: PropTypes.func.isRequired,
    activeViewport: PropTypes.string.isRequired,
  };

  render() {
    const { activeViewport, onChange } = this.props;
    return (
      <div style={styles.row}>
        <label style={styles.label}>Device</label>

        <select style={styles.action} value={activeViewport} onChange={onChange}>
          <option value={defaultViewport}>Default</option>

          {Object.keys(viewports).map(key =>
            <option value={key} key={key}>
              {viewports[key].name}
            </option>
          )}
        </select>
      </div>
    );
  }
}
