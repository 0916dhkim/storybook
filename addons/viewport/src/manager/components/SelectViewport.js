import React from 'react';
import PropTypes from 'prop-types';

import { Label, Row, Select } from './styles';

export function SelectViewport({ viewports, defaultViewport, activeViewport, onChange }) {
  return (
    <Row>
      <Label htmlFor="device">Device</Label>
      <Select id="device" value={activeViewport} onChange={onChange}>
        {Object.entries(viewports).map(([key, { name }]) => (
          <option value={key} key={key}>
            {key === defaultViewport ? `(Default) ${name}` : name}
          </option>
        ))}
      </Select>
    </Row>
  );
}

SelectViewport.propTypes = {
  onChange: PropTypes.func.isRequired,
  activeViewport: PropTypes.string.isRequired,
  viewports: PropTypes.shape({}).isRequired,
  defaultViewport: PropTypes.string.isRequired,
};
