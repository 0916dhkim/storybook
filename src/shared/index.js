// addons, panels and events get unique names using a prefix
export const ADDON_ID = 'kadirahq/storybook-addon-comments';
export const PANEL_ID = `${ADDON_ID}/comments-panel`;
