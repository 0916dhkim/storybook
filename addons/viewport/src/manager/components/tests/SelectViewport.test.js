import React from 'react';
import { shallow } from 'enzyme';
import { SelectViewport } from '../SelectViewport';
import { INITIAL_VIEWPORTS, DEFAULT_VIEWPORT } from '../../../shared';

const setup = () => {
  const props = {
    onChange: jest.fn(),
    activeViewport: DEFAULT_VIEWPORT,
    viewports: INITIAL_VIEWPORTS,
    defaultViewport: DEFAULT_VIEWPORT,
  };

  return { props, result: shallow(<SelectViewport {...props} />) };
};

describe('Viewport/SelectViewport', () => {
  it('is correctly rendered', () => {
    const { result } = setup();
    expect(result).toMatchSnapshot();
  });

  it('has a default option first', () => {
    const { result } = setup();
    expect(result.find('Styled(select)').props().value).toEqual(DEFAULT_VIEWPORT);
  });

  it('has at least 1 option', () => {
    const viewportKeys = Object.keys(INITIAL_VIEWPORTS);
    expect(viewportKeys.length).toBeGreaterThan(0);
  });

  const viewportKeys = Object.keys(INITIAL_VIEWPORTS);
  const { result } = setup();
  viewportKeys.forEach(key => {
    const { name } = INITIAL_VIEWPORTS[key];
    const expectedText = key === DEFAULT_VIEWPORT ? `(Default) ${name}` : name;

    it(`renders an option for ${name}`, () => {
      const option = result.find(`option[value="${key}"]`);
      expect(option.text()).toEqual(expectedText);
    });
  });
});
