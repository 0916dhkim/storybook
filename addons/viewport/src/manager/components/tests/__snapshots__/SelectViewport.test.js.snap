// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Viewport/SelectViewport is correctly rendered 1`] = `
<Styled(div)>
  <Styled(label)
    htmlFor="device"
  >
    Device
  </Styled(label)>
  <Styled(select)
    id="device"
    onChange={[MockFunction]}
    value="responsive"
  >
    <option
      key="responsive"
      value="responsive"
    >
      (Default) Responsive
    </option>
    <option
      key="iphone5"
      value="iphone5"
    >
      iPhone 5
    </option>
    <option
      key="iphone6"
      value="iphone6"
    >
      iPhone 6
    </option>
    <option
      key="iphone6p"
      value="iphone6p"
    >
      iPhone 6 Plus
    </option>
    <option
      key="ipad"
      value="ipad"
    >
      iPad
    </option>
    <option
      key="galaxys5"
      value="galaxys5"
    >
      Galaxy S5
    </option>
    <option
      key="nexus5x"
      value="nexus5x"
    >
      Nexus 5X
    </option>
    <option
      key="nexus6p"
      value="nexus6p"
    >
      Nexus 6P
    </option>
  </Styled(select)>
</Styled(div)>
`;
